app/controllers/products_controller.rb

```diff
@@ -0,0 +1,111 @@
+class ProductsController < ApplicationController
+  before_action :current_user
+  before_action :require_login, only: [:new, :create, :edit, :update, :retire]
+  before_action :check_user_product, only: [:review]
+  before_action :check_user_id, only: [:retire, :edit, :update]
+
+  def index
+    @categories = Category.all
+    @merchants = User.all
+    @products = Product.order(:created_at).where(retired: false)
+    case params[:type]
+
+      when "merch"
+        @products = @products.where("user_id = #{params[:order]}")
+      when "cat"
+        prod = Product.where(retired: false)
+        @products = prod.select {|product|
+          rows = product.categories.where("category_id = #{params[:order]}")
+          !rows.to_a.empty?
+        }
+      else
+        #binding.pry
+        @order = "prod"
+    end
+  end
+
+  def show
+    @product = Product.find(params[:id])
+    @review = Review.new
+    @reviews = Review.where("product_id = #{params[:id]}").reverse
```

Comment:
This kind of sorting logic to present the reviews in a reverse chronological way is great for users. Code-wise, it is model related, so should be in the model rather than the controller. If you need to pass in something accessible only from a controller, you can make that a parameter.